Add Slackware support to timezone module - Fix Issue 59130

piterpunk opened the pull request:
### What does this PR do?
Adds Slackware support to modules/timezone.py

This behaviour was catch by the timezone module integration tests.

Fixed `get_hwclock`, `set_hwclock` and `get_zone` functions.

### What issues does this PR fix or reference?
Fixes: #59130

### Previous Behavior
```
# salt-call timezone.get_hwclock
local:
    None
```

### New Behavior
```
# salt-call timezone.get_hwclock
local:
    UTC
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
- [ ] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

The bug was caught by the current integration tests as seen in #59130 . Now it pass without failures:
```
collected 5 items                                                                                                                  

tests/integration/modules/test_timezone.py .ssss

========================================================= warnings summary =========================================================
salt/ext/tornado/web.py:1805
  /home/punk/Desktop/Projects/Software/External/salt/salt/ext/tornado/web.py:1805: DeprecationWarning: invalid escape sequence \.
    """A collection of request handlers that make up a web application.

-- Docs: https://docs.pytest.org/en/stable/warnings.html
===================================================== short test summary info ======================================================
SKIPPED [1] tests/integration/modules/test_timezone.py:53: For Solaris only
SKIPPED [1] tests/integration/modules/test_timezone.py:69: windows test only
SKIPPED [1] tests/integration/modules/test_timezone.py:92: windows test only
SKIPPED [1] tests/integration/modules/test_timezone.py:74: Destructive tests are disabled
============================================= 1 passed, 4 skipped, 1 warning in 34.26s =============================================
```

### Commits signed with GPG?
No

Commit:
Fixed modules/timezone.py to work in Slackware

Commit:
Added changelog entry

s0undt3ch reviewed (changes requested):
Could we please have unit tests covering this change?

piterpunk commented:
> Could we please have unit tests covering this change?

Of course! I will write them and update this PR.

Commit:
Added timezone unit tests for Slackware

- Updated test_get_zone_os_family_allbsd_nilinuxrt to include Slackware
- Created test_set_zone_os_family_nilinuxrt for NILinuxRT
- Created test_set_zone_os_family_allbsd_slackware for FreeBSD, OpenBSD,
  NetBSD and Slackware
- Created tests from set_hwclock and get_hwclock for Slackware

piterpunk commented:
@s0undt3ch , added the unit tests.
